sgpt/utils.py

```diff
@@ -82,3 +86,20 @@ def echo_chat_ids() -> None:
     # Prints all existing chat IDs to the console.
     for chat_id in OpenAIClient.chat_cache.list():
         typer.echo(chat_id)
+
+
+def combine_with_stdin_prompt(prompt: str | None, stdin_before: bool) -> str | None:
+    stdinprompt: None | str = None
+    if select.select([sys.stdin], [], [], 0) != ([], [], []):
+        stdinprompt = sys.stdin.read().rstrip()
+
+    combined: None | str = None
+    if prompt is None and stdinprompt is None:
+        combined = None
+    elif prompt is None:
+        combined = stdinprompt
+    elif stdinprompt is None:
+        combined = prompt
+    else:
```

Comment:
Suggestion: If you write `not x` instead of `x is None` in these conditions, then it won't combine empty `""` prompts.

Example:

```
$ printf '' | sgpt prompt1
current_prompt: "prompt1
"
fixed_prompt: "prompt1"

# Similarly with
$ echo prompt2 | sgpt ''
```